locking/lockable.go

```diff
@@ -0,0 +1,209 @@
+package locking
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/git-lfs/git-lfs/config"
+	"github.com/git-lfs/git-lfs/errors"
+	"github.com/git-lfs/git-lfs/filepathfilter"
+	"github.com/git-lfs/git-lfs/git"
+	"github.com/git-lfs/git-lfs/tools"
+)
+
+// GetLockablePatterns returns a list of patterns in .gitattributes which are
+// marked as lockable
+func (c *Client) GetLockablePatterns() []string {
+	c.ensureLockablesLoaded()
+	return c.lockablePatterns
```

Comment:
A read-only mutex here wouldn't do anything useful unless the caller also locked it.

`ensureLockablesLoaded` ensures that `c.lockablePatterns` is only updated by one thread, and `refreshLockablePatterns` deliberately replaces the entire map (rather than clearing and re-populating it) so that previous callers to `GetLockablePatterns` can continue to use the old map. Therefore a read-only mutex is not required provided the return from `GetLockablePatterns` is not modified. Unfortunately Go doesn't have a `const` marker for returns, I thought about copying the map each time but ultimately thought it wasn't worth it.